From 30395e52d0b1c7e5bb93e387ea34a25218f10696 Mon Sep 17 00:00:00 2001
From: Soumyadeep Laskar <soumyadeeplaskar.2000@gmail.com>
Date: Sun, 31 May 2020 14:10:05 +0530
Subject: [PATCH] Create Patch File

---
 client3.py | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/client3.py b/client3.py
index f1771c3..4ec7ccb 100644
--- a/client3.py
+++ b/client3.py
@@ -29,22 +29,24 @@ QUERY = "http://localhost:8080/query?id={}"
 # 500 server request
 N = 500
 
-def getDataPoint(quote):
+def getDataPoint(quote): #Change here
 	""" Produce all of the needed values to generate a datapoint """
 	""" ------------- Update this function ------------- """
 	stock = quote['stock']
 	bid_price = float(quote['top_bid']['price'])
 	ask_price = float(quote['top_ask']['price'])
-	price = bid_price
+	price = (bid_price + ask_price)/
 	return stock, bid_price, ask_price, price
 
-def getRatio(price_a, price_b):
+def getRatio(price_a, price_b): #Change here
 	""" Get ratio of price_a and price_b """
 	""" ------------- Update this function ------------- """
 	""" Also create some unit tests for this function in client_test.py """
-	return 1
+	if(price_b == ):
+		return
+	return price_a/price_b
 
-# Main
+# Main #change here
 if __name__ == "__main__":
 
 	# Query the price once every N seconds.
@@ -52,8 +54,10 @@ if __name__ == "__main__":
 		quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
 		""" ----------- Update to get the ratio --------------- """
+		prices = {}
 		for quote in quotes:
 			stock, bid_price, ask_price, price = getDataPoint(quote)
+			prices[stock] = price
 			print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-		print ("Ratio %s" % getRatio(price, price))
+		print ("Ratio %s" % getRatio(prices['ABC'], prices['DEF']))
\ No newline at end of file
-- 
2.26.2.windows.1

